.github/workflows/main.yml

```diff
@@ -24,4 +24,35 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - run: docker build --target test --tag my-test-image .
-    - run: docker run my-test-image ./todo_app/tests
\ No newline at end of file
+    - run: docker run my-test-image ./todo_app/tests
+
+  deploy:
+    name: deploy
+    runs-on: ubuntu-latest
+    # if: github.ref == 'refs/heads/master'
+
+    needs: build
+    
+    steps:
+      - uses: actions/checkout@v2
+     
+      - name: Build Docker Image
+        run: docker build --target prod -t todo-app:latest .  
+      
+      - name: Log in to Docker Registry
+        run: echo "${{ secrets.DOCKER_REGISTRY_PSW }}" | docker login -u "${{ secrets.DOCKER_REGISTRY_USER }}" --password-stdin
+
+      - name: Tag the image
+        run: docker tag todo-app:latest "${{ secrets.DOCKER_REGISTRY_USER }}/todo-app:latest"
+
+      - name: Push Docker Image
+        run: docker push "${{ secrets.DOCKER_REGISTRY_USER }}/todo-app:latest"
+
+      - name: Update deploy image in Azure
+        env:
+          AZURE_WEBHOOK_URL: ${{ secrets.AZ_WH }}
+        run: |
+          AZURE_WEBHOOK_URL="${AZURE_WEBHOOK_URL//DOLLAR_SIGN/$}"
+          echo ${{ secrets.AZ_WH }}
+          curl -dH -X --fail POST '${{ secrets.AZ_WH }}' 
```

Comment:
This can get quite fiddly, so appreciate the comment saying this wasn't quite working!

The easiest way I find to be confident this will work is:
* Store the raw secret directly in GitHub's secret value, i.e. **don't** escape the $ symbol
* Convert the secret to an environment variable either at this step or at the job level, but make sure to wrap it in single quotes so it knows to interpret it as a literal
```yml
        env:
          AZURE_WEBHOOK_URL: '${{ secrets.AZ_WH }}'
```
* Pass it directly:
```bash
curl -dH -X --fail POST $AZURE_WEBHOOK_URL
```

Hopefully that will solve the issue!

Reply:
I wasted so much time because the flag --fail is in the wrong place so it should be: curl -dH -X --fail POST $AZURE_WEBHOOK_URL

So I started having problems I added --fail to debug, fix the problem but --fail was making it fail... anyway, now seems to work